packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -110,9 +110,12 @@ export function cleanupAfterRenderResources() {
 // from Internals -> ReactDOM -> FloatClient -> Internals so this doesn't introduce a new one.
 export const ReactDOMClientDispatcher = {preload, preinit};
 
-type FloatDocument = Document & {_rstyles?: Map<string, StyleResource>};
-type FloatShadowRoot = ShadowRoot & {_rstyles?: Map<string, StyleResource>};
-type FloatRoot = FloatDocument | FloatShadowRoot;
+const randomKey = Math.random()
+  .toString(36)
+  .slice(2);
+const stylesCacheKey = '__reactStyles$' + randomKey;
```

Comment:
This should also go into the same file as the others.

Reply:
So that we can change strategy, for example to WeakMaps or group multiple to one expando instead of several different ones for the same node etc.